pandas/core/frame.py

```diff
@@ -7029,7 +7039,7 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=False):
         from pandas.core.reshape.concat import concat
 
         if isinstance(other, (list, tuple)):
-            to_concat = [self] + other
+            to_concat = [self] + list(other)
```

Comment:
What error was this causing? I'm always hesitant to blanket convert tuples to lists as it can have non-trivial performance impacts

Reply:
Just to comment on this: Adding lists and tuples raises a TypeError. It wasn't caught in the tests, but mypy caught it. So a win for static type checking:-)